eng/SourceBuildPrebuiltBaseline.xml

```diff
@@ -11,6 +11,12 @@
     <UsagePattern IdentityGlob="System.Collections.Immutable/8.0.0" />
     <!-- Transitive dependency of Roslyn -->
     <UsagePattern IdentityGlob="System.Reflection.Metadata/8.0.0" />
+    <UsagePattern IdentityGlob="Microsoft.Build/17.7.2" />
```

Comment:
I believe this is caused because https://github.com/dotnet/roslyn/blob/a3c0cbd96fa27cded000dacb8114a7ab3d475420/eng/Directory.Packages.props#L39

In Roslyn, some projects needs to target .net7, and the last MSbuild(targeting .net 7) is 17.7.2
It bypassed the source build checks in Roslyn, when they shipped as Nuget packges.
Later the change flows to here and starts to cause the sourebuild breaks.